unitTests/ui/StarRatingTest.ts

```diff
@@ -0,0 +1,155 @@
+  const children = starRatingElement.children;
+  let numStars = 0;
+  let numActiveStars = 0;
+  let i = 0;
+  for (; i < children.length; i++) {
+    if (children.item(i).classList.contains(STAR_CSS_CLASS)) {
+      if (children.item(i).classList.contains(ACTIVE_STAR_CSS_CLASS)) {
+        numActiveStars++;
+      }
+      numStars++;
+    }
+  }
+  return { numStars: numStars, numActiveStars: numActiveStars };
+};
+
+export function StarRatingTest() {
+  describe('StarRating', () => {
+    let test: Mock.IBasicComponentSetup<StarRating>;
+
+    function initStarRatingComponent(ratingValue: String, numRatingsValue?: String, ratingScale?: number) {
+      const options: IStarRatingOptions = { rating: '@rating', numberOfRatings: '@numberOfRatings', ratingScale: ratingScale };
+      const result: IQueryResult = FakeResults.createFakeResult();
+
+      result.raw.rating = ratingValue;
+      result.raw.numberOfRatings = numRatingsValue;
+
+      test = Mock.advancedResultComponentSetup<StarRating>(StarRating, result, <Mock.AdvancedComponentSetupOptions>{
+        element: $$('div').el,
+        cmpOptions: options,
+        modifyBuilder: (builder: Mock.MockEnvironmentBuilder) => {
+          return builder;
+        }
+      });
+    }
+
+    describe('When the component is instantiated', () => {
+      it('should have correct class type', () => {
+        initStarRatingComponent('0', '0');
+        expect(test.cmp.element.classList.contains(CONTAINER_CSS_CLASS)).toBeTruthy;
+      });
+
+      describe('The star elements', () => {
+        it('should display five stars with a number of active stars equal to rating given', () => {
+          for (let i = 0; i <= 5; i++) {
+            initStarRatingComponent(i.toString());
+
+            let starData = getActiveStars(test.cmp.element);
+            expect(starData.numStars).toBe(5);
+            expect(starData.numActiveStars).toBe(i);
+          }
+        });
+
+        it('should display no active stars when the rating provided is negative', () => {
+          const testRating = -Number.MAX_VALUE;
+          initStarRatingComponent(testRating.toString());
+
+          let starData = getActiveStars(test.cmp.element);
+          expect(starData.numStars).toBe(5);
+          expect(starData.numActiveStars).toBe(0);
+        });
+      });
+
+      describe('The label showing number of ratings', () => {
+        it('should not display when no value is provided for number of ratings', () => {
+          initStarRatingComponent((0).toString());
+          const children = test.cmp.element.getElementsByClassName(STAR_LABEL_CSS_CLASS);
+          expect(children.length).toBe(0);
+        });
+
+        it('should display a label with "No Ratings" shown when zero given', () => {
+          const testNumRatings = 0;
+          initStarRatingComponent((0).toString(), testNumRatings.toString());
+
+          const testLabel = test.cmp.element.lastElementChild;
```

Comment:
A bit flaky that the test relies on the exact last element to have the correct class. Subtle modification to the component DOM would fail this test, even though it is still "okay".

Could simply query for the needed class, and check for the text equality.

```
const testLabel = $$(test.cmp.element).find(`.${STAR_LABEL_CSS_CLASS}`)
expect(testLabel).toBeDefined()
expect(testLabel.text()).toEqual(l('No Ratings'));
```